src/xcode/ENA/ENA/Source/Scenes/Checkin/Overview/CheckinsOverviewViewController.swift

```diff
@@ -0,0 +1,355 @@
+
+	private func setupTableView() {
+		tableView.register(
+			UINib(nibName: String(describing: AddEventTableViewCell.self), bundle: nil),
+			forCellReuseIdentifier: String(describing: AddEventTableViewCell.self)
+		)
+
+		tableView.register(
+			MissingPermissionsTableViewCell.self,
+			forCellReuseIdentifier: MissingPermissionsTableViewCell.reuseIdentifier
+		)
+
+		tableView.register(
+			UINib(nibName: String(describing: EventTableViewCell.self), bundle: nil),
+			forCellReuseIdentifier: String(describing: EventTableViewCell.self)
+		)
+
+		tableView.separatorStyle = .none
+		tableView.rowHeight = UITableView.automaticDimension
+		tableView.estimatedRowHeight = 60
+	}
+
+	private func animateChanges() {
+		DispatchQueue.main.async { [self] in
+			tableView.performBatchUpdates(nil, completion: nil)
+		}
+	}
+
+	private func updateRightBarButtonItem() {
+		let barButtonItem: UIBarButtonItem
+
+		if tableView.isEditing {
+			barButtonItem = editButtonItem
+		} else {
+			barButtonItem = UIBarButtonItem(
+				image: UIImage(named: "Icons_More_Circle"),
+				style: .plain,
+				target: self,
+				action: #selector(didTapMoreButton)
+			)
+			barButtonItem.accessibilityLabel = AppStrings.Checkins.Overview.menuButtonTitle
+			barButtonItem.tintColor = .enaColor(for: .tint)
+		}
+
+		parent?.navigationItem.setRightBarButton(barButtonItem, animated: true)
+	}
+
+	private func checkinAddCell(forRowAt indexPath: IndexPath) -> UITableViewCell {
+		guard let cell = tableView.dequeueReusableCell(withIdentifier: String(describing: AddEventTableViewCell.self), for: indexPath) as? AddEventTableViewCell else {
+			fatalError("Could not dequeue AddEventTableViewCell")
+		}
+
+		cell.configure(cellModel: addEntryCellModel)
+
+		return cell
+	}
+
+	private func missingPermissionsCell(forRowAt indexPath: IndexPath) -> UITableViewCell {
+		guard let cell = tableView.dequeueReusableCell(withIdentifier: MissingPermissionsTableViewCell.reuseIdentifier, for: indexPath) as? MissingPermissionsTableViewCell else {
+			fatalError("Could not dequeue MissingPermissionsTableViewCell")
+		}
+
+		cell.configure(
+			cellModel: missingPermissionsCellModel,
+			onButtonTap: { [weak self] in
+				self?.onMissingPermissionsButtonTap()
+			}
+		)
+
+		return cell
+	}
+
+	private func checkinCell(forRowAt indexPath: IndexPath) -> UITableViewCell {
+		guard let cell = tableView.dequeueReusableCell(withIdentifier: String(describing: EventTableViewCell.self), for: indexPath) as? EventTableViewCell else {
+			fatalError("Could not dequeue EventTableViewCell")
+		}
+
+		let cellModel = viewModel.checkinCellModels[indexPath.row]
+		cell.configure(
+			cellModel: cellModel,
+			onButtonTap: { [weak self] in
+				self?.viewModel.didTapEntryCellButton(at: indexPath)
+			}
+		)
+
+		return cell
+	}
+
+	private func updateEmptyState() {
+		tableView.backgroundView = viewModel.isEmpty ? EmptyStateView(viewModel: CheckinsOverviewEmptyStateViewModel()) : nil
```

Comment:
I know its convenient to use the backgroundView but it also leads to stuff like this:
![Simulator Screen Shot - iPhone 12 mini - 2021-03-18 at 18 10 22](https://user-images.githubusercontent.com/10122052/111667634-3633fe80-8815-11eb-8c75-d9ebfaa167c3.png) Would look even worse on smaller screens.